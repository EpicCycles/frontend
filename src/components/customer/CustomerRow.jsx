import React from "react";
import {Icon} from "semantic-ui-react";

const CustomerRow = props =>
    <tr key={`cust-row-${props.customer.id}`}>
        <td key={`cust-fn-${props.customer.id}`}>{props.customer.first_name}</td>
        <td key={`cust-ln-${props.customer.id}`}>{props.customer.last_name}</td>
        <td key={`cust-e-${props.customer.id}`}>{props.customer.email}</td>
        <td key={`cust-a-${props.customer.id}`}>{props.customer.add_date.substring(0,10)}</td>
        <td key={`cust-u-${props.customer.id}`}>{props.customer.upd_date.substring(0,10)}</td>
        <td key={`cust-act-${props.customer.id}`}>
            <Icon id={`edit-cust-${props.customer.id}`} name="edit"
                  onClick={() => props.getCustomer(props.customer.id)} title="Edit Customer"/>

        </td>
    </tr>;

export default CustomerRow;