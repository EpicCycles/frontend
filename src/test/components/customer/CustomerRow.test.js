import React from 'react';
import toJson from 'enzyme-to-json';
import CustomerRow from '../../../components/customer/CustomerRow';

describe('CustomerRow tests', () => {
  const customer = {
    id: 1,
    first_name: 'Anna',
    last_name: 'Weaver',
    email: 'anna.weaver@johnlewis.co.uk',
    add_date: '2018-07-04T13:02:09.988286+01:00',
    upd_date: '2018-07-04T13:02:09.988343+01:00',
  };

  it('renders the row correctly', () => {
    const input = shallow(<CustomerRow customer={customer} />);
    expect(toJson(input)).toMatchSnapshot();
  });

  it('calls the edit customer function when the button is clicked', () => {
    const editCustomer = jest.fn();

    const input = shallow(<CustomerRow customer={customer} getCustomer={editCustomer} />);
    expect(input.find('#edit-cust-1')).toHaveLength(1);
    input
      .find('#edit-cust-1')
      .at(0)
      .simulate('click');
    expect(editCustomer.mock.calls).toHaveLength(1);
  });
});
