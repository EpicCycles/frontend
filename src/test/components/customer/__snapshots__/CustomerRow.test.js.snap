// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerRow tests renders the row correctly 1`] = `
<tr
  key="cust-row-1"
>
  <td
    key="cust-fn-1"
  >
    Anna
  </td>
  <td
    key="cust-ln-1"
  >
    Weaver
  </td>
  <td
    key="cust-e-1"
  >
    anna.weaver@johnlewis.co.uk
  </td>
  <td
    key="cust-a-1"
  >
    2018-07-04
  </td>
  <td
    key="cust-u-1"
  >
    2018-07-04
  </td>
  <td
    key="cust-act-1"
  >
    <Icon
      as="i"
      id="edit-cust-1"
      name="edit"
      onClick={[Function]}
      title="Edit Customer"
    />
  </td>
</tr>
`;
